EIPS/eip-223.md

```diff
@@ -0,0 +1,368 @@
+
+Contracts intending to receive these tokens MUST implement `tokenReceived`.
+
+Tokens transferred to contracts not implementing `tokenReceived` as described below MUST revert.
+
+
+#### `totalSupply`
+
+```solidity
+function totalSupply() view returns (uint256 totalSupply)
+```
+
+Get the total token supply
+
+#### `name`
+
+```solidity
+function name() view returns (string _name)
+```
+
+Get the name of token
+
+#### `symbol`
+
+```solidity
+function symbol() view returns (bytes32 _symbol)
+```
+
+Get the symbol of token
+
+#### `decimals`
+
+```solidity
+function decimals() view returns (uint8 _decimals)
+```
+
+Get decimals of token
+
+#### `standard`
+
+```solidity
+function standard() view returns (string _standard)
+```
+
+Get the standard of token contract. For some services it is important to know how to treat this particular token. If token supports [ERC-223](./eip-223.md) standard then it must explicitly tell that it does.
+
+This function **MUST** return "ERC-223" for this token standard. If no "standard()" function is implemented in the contract then the contract must be considered to be ERC-20.
+
+#### `balanceOf`
+
+```solidity
+function balanceOf(address _owner) view returns (uint256 balance)
+```
+
+Get the account balance of another account with address `_owner`
+
+
+#### `transfer(address, uint)`
+
+```solidity
+function transfer(address _to, uint _value) returns (bool)
+```
+
+This function must transfer tokens and invoke the function `tokenReceived(address, uint256, bytes calldata)` in `_to`, if `_to` is a contract. If the `tokenReceived` function is not implemented in `_to` (receiver contract), then the transaction must fail and the transfer of tokens should be reverted.
+
+#### `transfer(address, uint, bytes)`
+
+```solidity
+function transfer(address _to, uint _value, bytes calldata _data) returns (bool)
+```
+
+function that is always called when someone wants to transfer tokens.
+This function must transfer tokens and invoke the function `tokenReceived (address, uint256, bytes)` in `_to`, if _to is a contract. If the `tokenReceived` function is not implemented in `_to` (receiver contract), then the transaction must fail and the transfer of tokens should not occur. 
+If `_to` is an externally owned address, then the transaction must be sent without trying to execute `tokenReceived` in `_to`.
+ `_data` can be attached to this token transaction and it will stay in blockchain forever (requires more gas). `_data` can be empty.
+
+NOTE: The recommended way to check whether the `_to` is a contract or an address is to assemble the code of `_to`. If there is no code in `_to`, then this is an externally owned address, otherwise it's a contract.
+
+### Events
+
+#### `Transfer`
+
+```solidity
+event Transfer(address indexed _from, address indexed _to, uint256 _value)
+```
+
+Triggered when tokens are transferred. Compatible with ERC-20 `Transfer` event.
+
+#### `TransferData`
+
+```solidity
+event TransferData(bytes _data)
+```
```

Comment:
Why not overload the `Transfer` event?

Reply:
Explorers and services that work with ERC20 tokens expect ERC20 Transfer event - `event Transfer(address indexed _from, address indexed _to, uint256 _value)`

From interface point of view both ERC20 and ERC223 look the same and they don't even need to dig it to understand what standard a token supports as both work the same.

So the idea is to keep it easier for interfaces

Reply:
Most libraries (including web3 and ethers) separate events per override. I see no feasible method for this to cause a backward incompatibility.

Reply:
https://github.com/Dexaran/EIPs/commit/3090cc6c3facc35d65ab67844d35e3c78f5a3aa3

Reply:
I don't understand.  Mind explaining?

Reply:
I have replaced `Transfer` and `TransferData` events with just one `Transfer` event with `bytes _data` parameter